adder/util.go

```diff
@@ -14,8 +14,20 @@ import (
 	rpc "github.com/libp2p/go-libp2p-gorpc"
 )
 
+type BlockPutHelper struct {
+	failedPeers map[peer.ID]struct{}
+	rpcClient   *rpc.Client
+}
+
+func NewBlockPutHelper(rpcClient *rpc.Client) *BlockPutHelper {
```

Comment:
is there anything preventing you from initializing with the destinations as well, and removing items from that slice as they fail (which saves you an amount of looping in every block)?